app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/dialogs/EnvironmentOptionsWidget.java

```diff
@@ -0,0 +1,202 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.vrbrowser.ui.widgets.dialogs;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.ScrollView;
+
+import org.mozilla.vrbrowser.R;
+import org.mozilla.vrbrowser.audio.AudioEngine;
+import org.mozilla.vrbrowser.browser.SessionStore;
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.ui.views.UIButton;
+import org.mozilla.vrbrowser.ui.views.settings.ButtonSetting;
+import org.mozilla.vrbrowser.ui.views.settings.RadioGroupSetting;
+import org.mozilla.vrbrowser.ui.views.settings.SwitchSetting;
+import org.mozilla.vrbrowser.ui.widgets.UIWidget;
+import org.mozilla.vrbrowser.ui.widgets.WidgetManagerDelegate;
+import org.mozilla.vrbrowser.ui.widgets.WidgetPlacement;
+import org.mozilla.vrbrowser.utils.LocaleUtils;
+
+public class EnvironmentOptionsWidget extends UIWidget implements
+        WidgetManagerDelegate.WorldClickListener,
+        WidgetManagerDelegate.FocusChangeListener {
+
+    private AudioEngine mAudio;
+    private UIButton mBackButton;
+
+    private SwitchSetting mEnvOverrideSwitch;
+    private RadioGroupSetting mEnvironmentsRadio;
+    private int mRestartDialogHandle = -1;
+
+    private ButtonSetting mResetButton;
+
+    private ScrollView mScrollbar;
+
+    public EnvironmentOptionsWidget(Context aContext) {
+        super(aContext);
+        initialize(aContext);
+    }
+
+    public EnvironmentOptionsWidget(Context aContext, AttributeSet aAttrs) {
+        super(aContext, aAttrs);
+        initialize(aContext);
+    }
+
+    public EnvironmentOptionsWidget(Context aContext, AttributeSet aAttrs, int aDefStyle) {
+        super(aContext, aAttrs, aDefStyle);
+        initialize(aContext);
+    }
+
+    private void initialize(Context aContext) {
+        inflate(aContext, R.layout.options_environment, this);
+
+        mAudio = AudioEngine.fromContext(aContext);
+
+        mWidgetManager.addFocusChangeListener(this);
+        mWidgetManager.addWorldClickListener(this);
+
+        mBackButton = findViewById(R.id.backButton);
+        mBackButton.setOnClickListener(view -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+
+            onDismiss();
+        });
+
+        String env = SettingsStore.getInstance(getContext()).getEnvironment();
+        mEnvironmentsRadio = findViewById(R.id.environmentRadio);
+        mEnvironmentsRadio.setOnCheckedChangeListener(mEnvsListener);
```

Comment:
UX request: would it be possible to increase the radius of the click target for the radio such that clicking anywhere on the row (the thumbnail, the radio button, or anywhere around but not precisely on the text line) would trigger a checked-change event?

I found myself thinking the UI was broken, until I was more precise with my clicks.

![image](https://user-images.githubusercontent.com/203725/54202916-52ff0400-448e-11e9-8c5a-f46393325896.png)